### PR TITLE
feat(B-169): add alembic setup and guidelines

### PR DESCRIPTION
**Pull request by clyko**
# Add alembic setup and guidelines
### Current Situation
No migration management for database schema evolution.
### Proposed Solution
Set up alembic for managing database schema migrations. Configurations under /backend/alembic.ini and under /backend/alembic/env.py. Initial migration script under /backend/alembic/versions. Edited backend/fly.toml to run latest migration on deployment.
#### Implications
Changes workflow for handling model updates, described within /backend/alembic/README.
Also activated .github/workflows/check-backend-on-pr.yml and .github/workflows/check-frontend-on-pr.yml. Added lint and test job to deploy-backend.yml. Tests currently fail because of missing open ai key under github secrets and linting fails because of ruff findings. Needs to be resolved for future prs to be merged and for future backend deployments.
### Testing
Triggered pipelines by dummy changes. Tested migration script locally and on prod db. Deployed backend to check if migration scripts are executed. Connected to container via ssh to check alembic state.
### Reviewer Notes
Double check migration script locally. Maybe also access prod supabase and container via ssh (fly ssh console --app coach-ai) to run migration scripts manually to verify.

### CODE REVIEW

**Comment by clyko** [comment]
Tbd:
- Testing the migration script manually on the prod db
- Testing the migration script from the deployment pipeline
Besides:
- Executing the backend tests on the pr workflow works for some reason, but not in the deployment pipeline as the Open AI key is required for some of the tests and not set as a Github Secret for the workflows. I honestly do not know why it works for the pr workflow
- Linting errors are found for the backend project in both pipelines.
- We should block merging the pr until the pr checks have passed
- We should add a pr workflow for changes to the migration scripts and/or models in a future issue. There are already some minor checks in the pre-commit hook.

**Review by rautenrieb — CHANGES_REQUESTED** [review]
Looks good. I have left some comments. Also I have not reviewed the workflows yet.

File: backend/Dockerfile [hunk 1]
```diff
@@ -11,6 +11,9 @@ ENV PATH="/app/.venv/bin:$PATH"
 # Ref: https://docs.astral.sh/uv/guides/integration/docker/#caching
 ENV UV_LINK_MODE=copy
 
+# postgresql client
+RUN apt-get update && apt-get install -y --no-install-recommends postgresql-client && rm -rf /var/lib/apt/lists/*
```
[the remark this answers is not in the document]
Comment (clyko) [reply backend/Dockerfile:15]: This is required for the test-migrations.sh which is executed by a docker container test-migrations in docker-compose.ci.yml. That container uses the backend Dockerfile.

The test-migrations.sh needs the postgresql-client to create a separate test_migrations database

File: backend/app/data/populate_dummy_data.py [hunk 2]
```diff
@@ -21,22 +22,16 @@
 
 
 def populate_data() -> None:
-    with DBSession(engine) as db_session:
-        db_session.exec(text('CREATE EXTENSION IF NOT EXISTS vector'))  # type: ignore
-        db_session.commit()
+    print('Downgrading to base...')
+    subprocess.run(['uv', 'run', 'alembic', 'downgrade', 'base'], check=True)
```
[the remark this answers is not in the document]
Comment (clyko) [reply backend/app/data/populate_dummy_data.py:26]: Lets maybe talk about this in more depth. In my opinion it makes sense to communicate now when everybody is using these changes for the first time, that they should drop the volume first. But after that it I think it will be an edge case to modify the table outside the models. In that case I think we can expect that these lines will also be edited temporarily by the developer.

File: backend/test.txt
[the remark this answers is not in the document]
Comment (clyko) [reply backend/test.txt:1]: Yes I removed it, was just for testing. The difs to dev should be clean now

**Review by rautenrieb — APPROVED** [review]
Looks great! Ready to merge from my side.
